pyk/src/pyk/proof/implies.py

```diff
@@ -409,9 +409,13 @@ def lines(self) -> list[str]:
 
 class ImpliesProver(Prover[ImpliesProof, ImpliesProofStep, ImpliesProofResult]):
     proof: ImpliesProof
+    kcfg_explore: KCFGExplore
+
+    def shutdown(self) -> None:
```

Comment:
Is this an abstract method on `Prover`? If so, maybe `Prover` should be a `ContextManager`, and use `__exit__` instead.